packages/hash/api/src/model/knowledge/orgMembership.model.ts

```diff
@@ -0,0 +1,170 @@
+      entityId: entity.entityId,
+      version: entity.version,
+      entityTypeModel,
+      properties,
+    });
+  }
+
+  /**
+   * Get a workspace organization entity by its entity id.
+   *
+   * @param params.entityId - the entity id of the organization
+   */
+  static async getOrgMembershipById(
+    graphApi: GraphApi,
+    params: { entityId: string },
+  ): Promise<OrgMembershipModel | null> {
+    const entity = await EntityModel.getLatest(graphApi, {
+      // assumption: `accountId` of organizations is always the workspace account id
+      accountId: workspaceAccountId,
+      entityId: params.entityId,
+    });
+
+    if (
+      entity.entityTypeModel.schema.$id !==
+      WORKSPACE_TYPES.entityType.orgMembership.schema.$id
+    ) {
+      throw new Error(
+        `Entity with id ${params.entityId} is not a workspace org membership`,
+      );
+    }
+
+    return entity ? new OrgMembershipModel(entity) : null;
+  }
+
+  /**
+   * Get the org linked to the org membership.
+   */
+  async getOrg(graphApi: GraphApi): Promise<OrgModel> {
+    const { data: outgoingOrgLinks } = await graphApi.getLinksByQuery({
+      all: [
+        {
+          eq: [{ path: ["source", "id"] }, { literal: this.entityId }],
+        },
+        {
+          eq: [
+            { path: ["type", "versionedUri"] },
+            {
+              literal: WORKSPACE_TYPES.linkType.ofOrg.schema.$id,
+            },
+          ],
+        },
+      ],
+    });
+
+    const [outgoingOrgLink] = outgoingOrgLinks;
+
+    if (!outgoingOrgLink) {
+      /**
+       * This should never be the case, as the `org` link is required on `OrgMembership` entities.
+       *
+       * @todo: potentially remove this when the Graph API validates entities based on their schema
+       */
+      throw new Error(
+        `Critical: org membership with entity id ${this.entityId} doesn't have an outgoing "org" link`,
+      );
+    }
+
+    const { targetEntityId: orgEntityId } = outgoingOrgLink;
+
+    const orgModel = await OrgModel.getOrgById(graphApi, {
+      entityId: orgEntityId,
+    });
+
+    if (!orgModel) {
+      throw new Error(
+        `Critical: org membership with entity id ${this.entityId} links to non-existent org with id ${orgEntityId}`,
+      );
+    }
+
+    return orgModel;
+  }
+
+  /**
+   * Get the user linked to the org membership.
+   */
+  async getUser(graphApi: GraphApi) {
+    const { data: incomingOrgMembershipLinks } = await graphApi.getLinksByQuery(
+      {
+        all: [
+          {
+            eq: [{ path: ["target", "id"] }, { literal: this.entityId }],
+          },
+          {
+            eq: [
+              { path: ["type", "versionedUri"] },
+              {
+                literal: WORKSPACE_TYPES.linkType.hasMembership.schema.$id,
+              },
+            ],
+          },
+        ],
+      },
+    );
+
+    const [incomingOrgMembershipLink] = incomingOrgMembershipLinks;
+
+    if (!incomingOrgMembershipLink) {
```

Comment:
Clarification question: Is this the same as `incomingOrgMembershipLink === undefined`? Is it possible, that a `Link` can be converted to a boolean? If there is any chance this could happen, we should use an explicit comparison.

Reply:
It's more this https://developer.mozilla.org/en-US/docs/Glossary/Falsy

Reply:
> Is this the same as incomingOrgMembershipLink === undefined?

Not exactly, it would also account for if `incomingOrgMembershipLink` is `null` for example - but it achieves the same behaviour in this example yes (because `incomingOrgMembershipLink` can only be `Link` or `undefined`.

Reply:
> Is it possible, that a Link can be converted to a boolean?

I'm not sure what you mean, `incomingOrgMembershipLink` should only have type `Link` or be `undefined`. When would `Link` be converted to a boolean?